### PR TITLE
Add default FileSystemWatcher to unityaot profile (case 1344045).

### PR DESCRIPTION
**Pull request by joncham**
Fix FileSystemWatcher when running on IL2CPP.

- Should this pull request have release notes?
  - [x] Yes
  - [ ] No
- Do these changes need to be back ported?
  - [x] Yes
  - [ ] No
- Do these changes need to be upstreamed to [mono/mono](https://github.com/mono/mono) or [dotnet/runtime](https://github.com/dotnet/runtime) repositories?
  - [ ] Yes
  - [x] No

Reviewers: please consider these questions as well! :heart:

**Release notes**

Fixed case 1344045 @joncham:
Mono: Fix FileSystemWatcher support on IL2CPP.

**Backports**

2021.2

### CODE REVIEW

File: mcs/class/System/linux_unityaot_System.dll.sources [hunk 1]
```diff
@@ -3,4 +3,10 @@
...
 ../System.Web/System.Web.Util/HttpEncoder.cs
 System.CodeDom/CodeCompileUnit.cs
-System.CodeDom/CodeTypeDeclaration.cs
\ No newline at end of file
+System.CodeDom/CodeTypeDeclaration.cs
+
+#include unityaot_fsw.sources
+
+../../../external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs
+../../../external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.UnknownUnix.cs
```
Comment (bholmes) [inline mcs/class/System/linux_unityaot_System.dll.sources]: add the 2 cs files to `unityaot_fsw.sources` ?

```
../../../external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs
../../../external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.UnknownUnix.cs
```

Comment (bholmes) [reply mcs/class/System/linux_unityaot_System.dll.sources]: same for macos and win32 files

File: external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Enumeration;
+using System.Threading;
+using System.Threading.Tasks;
+
+#if MONO && (!MOBILE || MOBILE_DESKTOP_HOST || UNITY_AOT)
```
Comment (bholmes) [inline external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs:15]: Did you make any other change in this file?  Did you annotate it?

**Review by bholmes — CHANGES_REQUESTED** [review]

File: mcs/class/System/linux_unityaot_System.dll.sources (hunk 1 above)
Comment (joncham) [reply mcs/class/System/linux_unityaot_System.dll.sources]: I can fix this.

File: external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs (hunk 2 above)
Comment (joncham) [reply external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs:15]: I had done this as separate commit but forgot and squashed. Let me diff and verify.

**Comment by UnityAlex** [comment]
Looks good so far just waiting on the diff for `FileSystemWatcher.cs`

File: mcs/class/System/linux_unityaot_System.dll.sources (hunk 1 above)
Comment (joncham) [reply mcs/class/System/linux_unityaot_System.dll.sources]: Force pushed a fix for this.

File: external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs (hunk 2 above)
Comment (joncham) [reply external/corefx-bugfix/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs:15]: Pushed as two separate commits to see changes.

**Comment by joncham** [comment]
> Looks good so far just waiting on the diff for `FileSystemWatcher.cs`

Pushed

**Review by bholmes — APPROVED** [review]

**Review by UnityAlex — APPROVED** [review]

**Pull request merged**